File: padroesComportamentais/padrao-de-projeto-iterator/php/tests/LojaTest.php
<?php

namespace Iterator\tests;

use Iterator\Php\Carrinho;
use Iterator\Php\Loja;
use Iterator\Php\Produto;
use PHPUnit\Framework\TestCase;

class LojaTest extends TestCase
{
    public function testDeveRetornarValorTotalCarrinho(): void
    {
        $produto = (new Produto())->setNome('teste1')->setValor(10.0);
        $produto2 = (new Produto())->setNome('teste2')->setValor(15.0);
        $carrinho = (new Carrinho())
            ->adicionarProduto($produto)
            ->adicionarProduto($produto2);

        $loja = new Loja();

        $this->assertEquals(25.0, $loja->contaValorTotalCarrinho($carrinho), 0.01);
    }

    public function testDeveRetornarQuantidadeProdutoCarrinho(): void
    {
        $produto = (new Produto())->setNome('teste1')->setValor(10.0);
        $produto2 = (new Produto())->setNome('teste2')->setValor(15.0);
        $carrinho = (new Carrinho())
            ->adicionarProduto($produto)
            ->adicionarProduto($produto2);

        $loja = new Loja();

        $this->assertEquals(2, $loja->contaTotalProdutosCarrinho($carrinho));
    }
}

File: padroesComportamentais/padrao-de-projeto-iterator/java/src/test/java/LojaTest.java
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class LojaTest {

    @Test
    public void deveRetornarValorTotalCarrinho() {
        Produto produto = new Produto().setNome("teste").setValor(10f);
        Produto produto2 = new Produto().setNome("teste2").setValor(15f);
        Carrinho carrinho = new Carrinho().adicionaProduto(produto).adicionaProduto(produto2);
        Loja loja = new Loja();
        assertEquals(25f, loja.contaValorTotalCarrinho(carrinho), 0.01);
    }

    @Test
    public void deveRetornarQuantidadeProdutoCarrinho() {
        Produto produto = new Produto().setNome("teste").setValor(10f);
        Produto produto2 = new Produto().setNome("teste2").setValor(15f);
        Carrinho carrinho = new Carrinho().adicionaProduto(produto).adicionaProduto(produto2);
        Loja loja = new Loja();
        assertEquals(2, loja.contaTotalProdutosCarrinho(carrinho));
    }
}
